lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -1627,6 +1633,15 @@ void IRGenDebugInfoImpl::popLoc() {
   std::tie(LastDebugLoc, LastScope) = LocationStack.pop_back_val();
 }
 
+void IRGenDebugInfoImpl::setInlinedTrapLocation(IRBuilder &Builder,
+                                                const SILDebugScope *Scope) {
+  auto DLInlinedAt = llvm::DebugLoc::get(LastDebugLoc.Line, LastDebugLoc.Column,
+                                         getOrCreateScope(LastScope));
+  // FIXME: This location should point to stdlib instead of being artificial.
+  auto DL = llvm::DebugLoc::get(0, 0, getOrCreateScope(Scope), DLInlinedAt);
```

Comment:
Making the ``llvm.trap`` instruction have an artificial location while also being ``inlinedAt`` the ``+`` works well with WinDbg. I would have liked to get the location of ``llvm.trap`` in ``Integers.swift``, but it looks like it doesn't even have a ``SouceLoc`` before ``MandatoryInline``.

Reply:
Yeah, we don't preserve locations across module boundaries today. I wouldn't worry too much about it as long as the user can clearly get back to the `+` that they wrote when the overflow happens.

Reply:
(Still learning how to use GitHub. Here is my last comment) Ok, that explains a lot. I have tested integer overflow with WinDbg and it reports the addition to be where the trap occurred.